specs/_features/eip7251/beacon-chain.md

```diff
@@ -429,6 +440,29 @@ def set_compounding_withdrawal_credentials(state: BeaconState, index: ValidatorI
         validator.withdrawal_credentials[:1] = COMPOUNDING_WITHDRAWAL_PREFIX
 ```
 
+#### New `switch_to_compounding_validator`
+
+```python
+def switch_to_compounding_validator(state: BeaconState, index: ValidatorIndex) -> None:
+    validator = state.validators[index]
+    if has_eth1_withdrawal_credential(validator):
+        validator.withdrawal_credentials[:1] = COMPOUNDING_WITHDRAWAL_PREFIX
+        queue_excess_active_balance(state, index)
+```
+
+#### New `queue_excess_active_balance`
+
+```python
+def queue_excess_active_balance(state: BeaconState, index: ValidatorIndex) -> None:
+    balance = state.balances[index]
+    if balance > MAX_EFFECTIVE_BALANCE:
+        excess_balance = balance - MAX_EFFECTIVE_BALANCE
+        state.balances[index] = balance - excess_balance
```

Comment:
feels clearer to just say:
```
state.balances[index] = MAX_EFFECTIVE_BALANCE
```

Reply:
```suggestion
        state.balances[index] = MAX_EFFECTIVE_BALANCE
```

It feels confusing to use both `MAX_EFFECTIVE_BALANCE` and `MAX_EFFECTIVE_BALANCE_EIP7251` in the same file. What do you think about adding a duplicate `MAX_EFFECTIVE_BALANCE_PRE_EIP7251 := 32`?

Reply:
I agree with that, maybe `MAX_EFFECTIVE_BALANCE_PHASE0`?

Reply:
oh yeah using `MIN_ACTIVATION_BALANCE` looks better to me 👍

Reply:
So you suggest the opposite, to replace `MAX_EFFECTIVE_BALANCE` with `MIN_ACTIVATION_BALANCE` everywhere in the spec? I am fine with either, as long as we use the same param name in all places

Reply:
If the two options are replacing `MIN_ACTIVATION_BALANCE` with `MAX_EFFECTIVE_BALANCE_PHASE0` or replacing `MAX_EFFECTIVE_BALANCE` with `MIN_ACTIVATION_BALANCE`, I would prefer the latter, because `MIN_ACTIVATION_BALANCE` accurately represents what this parameter is in the current system

Reply:
> because `MIN_ACTIVATION_BALANCE` accurately represents what this parameter is in the current system

Not in every case, it is also a max effective balance for 0x01 creds. And when we want to get MAXEB of a particular validator it makes more sense to use `MAX_EFFECTIVE_BALANCE_PHASE) and *_EIP7251`. But again, I am fine with either, just don’t want to have `MAX_EB`, `MIN_AB` and `MAX_EB_EIP7251` in the spec as this is really confusing.

Reply:
For bumped constants in the past we have used: 

```
INACTIVITY_PENALTY_QUOTIENT
INACTIVITY_PENALTY_QUOTIENT_ALTAIR
INACTIVITY_PENALTY_QUOTIENT_BELLATRIX
```